### PR TITLE
SCLI-35 SPI discards GLT response and stops txn recovery

### PR DESCRIPTION
**Pull request by metinavc1**

### CODE REVIEW

File: Library/SPIClient.m [hunk 1]
```diff
@@ -1201,7 +1201,6 @@ - (void)handleGetLastTransactionResponse:(SPIMessage *)m {
             } else {
                 // TH-4X - Unexpected Error when recovering
                 SPILog(@"Unexpected response in get last transaction during - received posRefId:%@ error: %@", [gltResponse getPosRefId], m.error);
```
Comment (ellulpatrick) [inline Library/SPIClient.m]: can you perhaps amend the logged message to say that we are ignoring it?
`SPILog(@"Unexpected response in get last transaction during - received posRefId:%@ error: %@. Ignoring.", [gltResponse getPosRefId], m.error);`

File: Library/SPIClient.m [hunk 2]
```diff
@@ -1201,7 +1201,6 @@ - (void)handleGetLastTransactionResponse:(SPIMessage *)m {
...
-                [txState unknownCompleted:@"Unexpected error when recovering transaction status. Check EFTPOS."];
```
Comment (ellulpatrick) [inline Library/SPIClient.m:1204]: you need to replace this with a `return` statement so we don't unnecessarily   publish the state change.

**Review by ellulpatrick — CHANGES_REQUESTED** [review]
Hi Metin. just a couple of things.

File: Library/SPIClient.m (hunk 1 above)
Comment (metinavc1) [reply Library/SPIClient.m]: Added it

File: Library/SPIClient.m (hunk 2 above)
Comment (metinavc1) [reply Library/SPIClient.m:1204]: Added.

**Review by ellulpatrick — APPROVED** [review]
nice. thanks

**Pull request merged**